src/routes/users.js

```diff
@@ -10,12 +10,15 @@ import {
 import {User} from '../mongoose';
 
 export const authUser = async (req, res) => {
-  const {email, password} = req?.body;
+  const {email, password} = req.body;
 
   await User.findOne({email})
     .then((userDoc) => {
       if (!userDoc) {
-        return handleNotFound(res);
+        handleNotFound(res);
+      }
+      if (!password) {
+        res.status(400).send('Please provide a valid password.');
```

Comment:
Nice catch! Can you move this check to before we run the User.findOne query so that we dont run an unnecessary query?